signinwithapplebutton/src/main/java/com/willowtreeapps/signinwithapplebutton/SignInWithAppleService.kt

```diff
@@ -34,36 +54,68 @@ class SignInWithAppleService(
+            which is why certain fields (like the version, v) are included in the URI construction.
+
+            We have to build this URI ourselves because Apple's behavior in JavaScript is to POST the response,
+            while we need a GET so we can retrieve the authentication code and verify the state
+            merely by intercepting the URL.
+
+            See the Sign In With Apple Javascript SDK for comparison:
+            https://developer.apple.com/documentation/signinwithapplejs/configuring_your_webpage_for_sign_in_with_apple
+            */
+            fun create(
+                config: SignInWithAppleConfig,
+                state: String = UUID.randomUUID().toString()
+            ): AuthenticationAttempt {
+                val authenticationUri = Uri
+                    .parse("https://appleid.apple.com/auth/authorize")
+                    .buildUpon().apply {
+                        appendQueryParameter("response_type", "code")
+                        appendQueryParameter("v", "1.1.6")
+                        appendQueryParameter("client_id", config.clientId)
+                        appendQueryParameter("redirect_uri", config.redirectUri)
+                        appendQueryParameter("scope", config.scope)
+                        appendQueryParameter("state", state)
+                    }
+                    .build()
+                    .toString()
+
+                return AuthenticationAttempt(authenticationUri, config.redirectUri, state)
+            }
         }
     }
 
-    /*
-    The authentication page URI we're creating is based off the URI constructed by Apple's JavaScript SDK,
-    which is why certain fields (like the version, v) are included in the URI construction.
-
-    We have to build this URI ourselves because Apple's behavior in JavaScript is to POST the response,
-    while we need a GET so we can retrieve the authentication code and verify the state
-    merely by intercepting the URL.
-
-    See the Sign In With Apple Javascript SDK for comparison:
-    https://developer.apple.com/documentation/signinwithapplejs/configuring_your_webpage_for_sign_in_with_apple
-    */
-    internal fun buildAuthenticationAttempt(state: String = UUID.randomUUID().toString()): AuthenticationAttempt {
-        val authenticationUri = Uri
-            .parse("https://appleid.apple.com/auth/authorize")
-            .buildUpon().apply {
-                appendQueryParameter("response_type", "code")
-                appendQueryParameter("v", "1.1.6")
-                appendQueryParameter("client_id", clientId)
-                appendQueryParameter("redirect_uri", redirectUri)
-                appendQueryParameter("scope", scope)
-                appendQueryParameter("state", state)
-            }
-            .build()
-            .toString()
+    fun show() {
+        val fragment = SignInWebViewDialogFragment.newInstance(AuthenticationAttempt.create(config))
```

Comment:
One of these static object creating methods is `newInstance` and the other is `create` … is there anything idiomatic about either one that we can standardize with, or are they different for a reason?

Reply:
So there is no common convention here. `newInstance()` is common for fragments as that's used in the docs, but I've seen things like `create()` and `of()` used for factory methods. This was on an internal class so I didn't put too much thought into it, but I'm fine using `newInstance()` consistently here.

Reply:
Use your judgement; I'm just looking to learn about what's idiomatic and what's personal taste